### DISCUSSION TITLE
JWEDecryptionFailed: decryption operation failed

### DISCUSSION THREAD

**Discussion by therealsqvad** [panva/jose#82] [Q&A]
Hello everyone.
You need to decrypt the JWE and get the data.
Checked through JAVA - using the existing key it is possible to decrypt the message, but in nodejs it does not work
The library gives an error
_JWEDecryptionFailed: decryption operation failed_
the key is parsed, the message is validated by JWE, it already crashes during decryption

An error does not occur in the function at the time of return
aes_gcm.js
```
const decrypt = (size, {[KEYOBJECT]: keyObject}, ciphertext, {iv, tag = Buffer.alloc (0), aad = Buffer.alloc (0)}) => {
  const key = keyObject.asInput? keyObject.asInput (): keyObject
  checkInput (size, iv, tag)

  try {
    const cipher = createDecipheriv (`aes - $ {size} -gcm`, key, iv)
    cipher.setAuthTag (tag)
    cipher.setAAD (aad)

    **return Buffer.concat ([cipher.update (ciphertext), cipher.final ()])**
  } catch (err) {
    throw new JWEDecryptionFailed ()
  }
}
```

My code
```
const key = fs.readFileSync(keyFile);
const enc = encryptedData
try {
  const key = jose.JWK.asKey(keyData);
  jose.JWE.decrypt(
    enc,
    key,
  );
} catch (error) {
  console.log('error: ', error);```

**Comment by panva** [comment]
Please provide something tangible - an actual JWE token, the secret key, the java code that produced it, your node version, your library version, etc

**Comment by therealsqvad** [comment]
Unfortunately, JAVA code that can decrypt under NDA
And the existing JWE contains sensitive data

**Comment by panva** [comment]
In such case I cannot help you. All I can say that this implementation is checked against the [IETF RFC7520](https://tools.ietf.org/html/rfc7520) test vectors (includes 5 `aes-...-gcm` cases) so it is likely issue exists in your java code - and it only makes sense that since it produced the JWE it can also decrypt, but it's likely doing something wrong. Is that a java library you're using or proprietary implementation?

Give me _something_ to work with here. Best would be the producing code. I don't see how a snippet of code that encrypts data can be under NDA. But it baffles you expect to get help without being able to provide anything reproducible.

**Comment by therealsqvad** [comment]
[Here is the decryption function in JAVA](https://github.com/PaylinePaymentMethod/payment-method-samsung-pay/blob/79802feb728241912a4a4747221bae87ac118973/src/main/java/com/payline/payment/samsung/pay/utils/JweDecrypt.java), which can be decrypted using the key. The encryption function is not available to me

**Comment by panva** [comment]
That's of no use to me. I just checked [Nimbus](https://connect2id.com/products/nimbus-jose-jwt) produced JWEs and AES GCM works with no issues, ergo - this library is interoperable with other JWE conform libraries. Your JWE producer is likely implementing JWE wrong.
